src/yarp.c

```diff
@@ -1977,6 +1979,35 @@ yp_interpolated_xstring_node_closing_set(yp_interpolated_x_string_node_t *node,
   node->base.location.end = closing->end;
 }
 
+// Allocate a new KeywordHashNode node.
+static yp_keyword_hash_node_t *
+yp_keyword_hash_node_create(yp_parser_t *parser) {
+  yp_keyword_hash_node_t *node = yp_alloc(parser, sizeof(yp_keyword_hash_node_t));
+
+  *node = (yp_keyword_hash_node_t) {
+    .base = {
+      .type = YP_NODE_KEYWORD_HASH_NODE,
+      .location = {
+        .start = NULL,
+        .end = NULL
+      },
+    }
+  };
+
+  yp_node_list_init(&node->elements);
+  return node;
+}
+
+// Append an element to a KeywordHashNode node.
+static void
+yp_keyword_hash_node_elements_append(yp_keyword_hash_node_t *hash, yp_node_t *element) {
+  yp_node_list_append(&hash->elements, element);
+  if (hash->elements.size == 1) {
```

Comment:
Is this check more straightforward as `hash->base.location.start == NULL`?

Reply:
Yup, they are definitely logically the same, I think just more clear to read with the NULL check, thanks!